SonarQube.Client/Requests/PagedRequestBase.cs

```diff
@@ -60,7 +63,8 @@ public async override Task<TResponseItem[]> InvokeAsync(HttpClient httpClient, C
 
                 Page++;
             }
-            while (pageResult.Value != null &&
+            while (!ReachedMaxPageNumber() &&
+                pageResult.Value != null &&
```

Comment:
@duncanp-sonar do you want to fix https://github.com/SonarSource/sonarqube-webclient-dotnet/issues/8 as well?

Reply:
As discussed offline, changing this check to be "max issues reached" might fix the other issue too, but we'd need to check that the behaviour of e.g. `GetQualityProfile` isn't broken.